libraries/chain/block_log.cpp

```diff
@@ -628,6 +661,12 @@ namespace eosio { namespace chain {
    signed_block_ptr block_log::read_block_by_num(uint32_t block_num)const {
       try {
          signed_block_ptr b;
+
+         if (my->not_generate_block_log) {
+            // No blocks exist. Avoid cascading failures if going further.
+            return b;
+         }
```

Comment:
There are a number of public methods of `block_log` that should also check this flag. Including: read_block, read_block_header, read_block_by_num, read_block_id_by_num, read_block_by_id, get_block_pos, read_head, flush, etc. These should all be updated to do the correct thing when in `not_generate_block_log` mode.

Reply:
Ideally we should  have unittests around this new functionality so we can show all these work in the new mode.